<?xml version="1.0"?>
<launch>
    <arg name="camera_id" default="alexa"/>
    <arg name="topic_adapter" default="/realm/$(arg camera_id)/input"/>
    <arg name="topic_pose_est" default="/realm/$(arg camera_id)/pose_estimation/frame"/>
    <arg name="topic_dense" default="/realm/$(arg camera_id)/densification/frame"/>
    <arg name="topic_surf" default="/realm/$(arg camera_id)/surface_generation/frame"/>
    <arg name="topic_rect" default="/realm/$(arg camera_id)/ortho_rectification/frame"/>
    <arg name="topic_mosaic" default="/realm/$(arg camera_id)/mosaicing/frame"/>

    <node pkg="realm_ros" type="realm_exiv2_grabber" name="realm_exiv2_grabber" output="screen">
        <param name="config/id" type="string" value="$(arg camera_id)"/>
        <param name="config/input" type="string" value="/media/lachs/2aba9048-bfb9-4c30-9a82-d9b005f6ac2c/Entwicklung/Bilddaten/wabco/2_70m"/>
        <param name="config/rate" type="double" value="1.0"/>
        <param name="config/profile" type="string" value="phantom_IFF"/>
        <!--param name="config/opt/poses" type="string" value=""/-->
        <!--param name="config/opt/surface_pts" type="string" value=""/-->
        <!--param name="config/opt/set_all_keyframes" type="bool" value="True"/-->
    </node>

    <node pkg="realm_ros" type="realm_stage_node" name="realm_pose_estimation" output="screen">
        <param name="stage/type" type="string" value="pose_estimation"/>
        <param name="topics/input" type="string" value="$(arg topic_adapter)"/>
        <param name="topics/output" type="string" value="$(arg topic_pose_est)"/>
        <param name="config/id" type="string" value="$(arg camera_id)"/>
        <param name="config/profile" type="string" value="phantom_IFF"/>
        <param name="config/method" type="string" value="orb_slam2"/>
    </node>

    <node pkg="realm_ros" type="realm_stage_node" name="realm_densification" output="screen">
        <param name="stage/type" type="string" value="densification"/>
        <param name="topics/input" type="string" value="$(arg topic_pose_est)"/>
        <param name="topics/output" type="string" value="$(arg topic_dense)"/>
        <param name="config/id" type="string" value="$(arg camera_id)"/>
        <param name="config/profile" type="string" value="phantom_IFF"/>
        <param name="config/method" type="string" value="psl"/>
    </node>

    <node pkg="realm_ros" type="realm_stage_node" name="realm_surface_generation" output="screen">
        <param name="stage/type" type="string" value="surface_generation"/>
        <param name="topics/input" type="string" value="$(arg topic_dense)"/>
        <param name="topics/output" type="string" value="$(arg topic_surf)"/>
        <param name="config/id" type="string" value="$(arg camera_id)"/>
        <param name="config/profile" type="string" value="phantom_IFF"/>
    </node>

    <node pkg="realm_ros" type="realm_stage_node" name="realm_ortho_rectification" output="screen">
        <param name="stage/type" type="string" value="ortho_rectification"/>
        <param name="topics/input" type="string" value="$(arg topic_surf)"/>
        <param name="topics/output" type="string" value="$(arg topic_rect)"/>
        <param name="config/id" type="string" value="$(arg camera_id)"/>
        <param name="config/profile" type="string" value="phantom_IFF"/>
    </node>

    <node pkg="realm_ros" type="realm_stage_node" name="realm_mosaicing" output="screen">
        <param name="stage/type" type="string" value="mosaicing"/>
        <param name="stage/master" type="bool" value="true"/>
        <param name="topics/input" type="string" value="$(arg topic_rect)"/>
        <param name="config/id" type="string" value="$(arg camera_id)"/>
        <param name="config/profile" type="string" value="phantom_IFF"/>
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find realm_ros)/rviz/realm_test.rviz" />



</launch>
